scripts/src/index.ts

```diff
@@ -0,0 +1,199 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+import algoliasearch from "algoliasearch";
+
+import simulationMap from "../../resources/docs/url_map.json";
+
+function uuidv4() {
+  return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function (c) {
+    var r = (Math.random() * 16) | 0,
+      v = c == "x" ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+  });
+}
+
+const addUuidGlossary = () => {
+  const files = fs.readdirSync("../resources/glossary");
+
+  for (const fileName of files) {
+    const file = fs.readFileSync(`../resources/glossary/${fileName}`, "utf8");
+
+    fs.writeFileSync(
+      `./output/glossary/${fileName}`,
+      file.replace("---", `---\nobjectId: ${uuidv4()}`)
+    );
+  }
+};
+
+const getAllFiles = function (
+  dirPath,
+  arrayOfFiles: Array<{ inputPath: string; outputPath: string }>
+) {
+  const files = fs.readdirSync(dirPath);
+
+  arrayOfFiles = arrayOfFiles || [];
+
+  files.forEach(function (file) {
+    if (file === "LICENSE.MD") {
+      return;
+    }
+
+    if (fs.statSync(dirPath + "/" + file).isDirectory()) {
+      arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles);
+    } else {
+      const inputPath = path.join(dirPath, "/", file);
+      const outputPath = `.\\output\\${path.join(
+        "./output/",
+        dirPath,
+        "/",
+        file
+      )}`;
+      arrayOfFiles.push({ inputPath, outputPath });
+    }
+  });
+
+  return arrayOfFiles;
+};
+
+function ensureDirectoryExistence(filePath) {
+  var dirname = path.dirname(filePath);
+  if (fs.existsSync(dirname)) {
+    return true;
+  }
+  ensureDirectoryExistence(dirname);
+  fs.mkdirSync(dirname);
+}
+
+const addUuidDocs = () => {
+  const files = getAllFiles("../resources/docs/simulation", []);
+
+  for (const file of files) {
+    writeFile(file.inputPath, file.outputPath);
+  }
+};
+
+const writeFile = (
+  inputPath,
+  outputPath,
+  textData = `objectId: ${uuidv4()}`
+) => {
+  const file = fs.readFileSync(inputPath, "utf8");
+
+  ensureDirectoryExistence(outputPath);
+
+  if (file.startsWith("---")) {
+    fs.writeFileSync(outputPath, file.replace("---", `---\n${textData}`));
+  } else {
+    fs.writeFileSync(outputPath, `---\n${textData}}\n---\n\n${file}`);
+  }
+};
+
+const writeDocsFile = (urlObj) => {
+  let inputPath = `../resources/docs/${urlObj.slug}.md`;
+  let outputPath = `./output/${urlObj.slug}.md`;
+
+  if (!fs.existsSync(inputPath)) {
+    inputPath = `../resources/docs/${urlObj.slug}/README.md`;
+    outputPath = `./output/${urlObj.slug}/README.md`;
+  }
+
+  writeFile(
+    inputPath,
+    outputPath,
+    `title: ${urlObj.title}\nslug: ${urlObj.slug}\nobjectId: ${uuidv4()}`
+  );
+};
+
+const addTitleToPage = () => {
```

Comment:
This and a few other functions appear to be for data migration rather than repeated use - do we need to keep them all?

Reply:
I thought they would be useful, but yeah, they can remain in git history